File: old_samples/sample-chat/src/main/java/com/quickblox/sample/chat/utils/qb/QbUsersHolder.java
package com.quickblox.sample.chat.utils.qb;

import android.util.SparseArray;

import com.quickblox.users.model.QBUser;

import java.util.ArrayList;
import java.util.List;

/**
 * Basically in your app you should store users in database
 * And load users to memory on demand
 * We're using runtime SpaceArray holder just to simplify app logic
 */
public class QbUsersHolder {

    private static QbUsersHolder instance;

    private SparseArray<QBUser> qbUserSparseArray;

    public static synchronized QbUsersHolder getInstance() {
        if (instance == null) {
            instance = new QbUsersHolder();
        }

        return instance;
    }

    private QbUsersHolder() {
        qbUserSparseArray = new SparseArray<>();
    }

    public void putUsers(List<QBUser> users) {
        for (QBUser user : users) {
            putUser(user);
        }
    }

    public void putUser(QBUser user) {
        qbUserSparseArray.put(user.getId(), user);
    }

    public QBUser getUserById(int id) {
        return qbUserSparseArray.get(id);
    }

    public List<QBUser> getUsersByIds(List<Integer> ids) {
        List<QBUser> users = new ArrayList<>();
        for (Integer id : ids) {
            QBUser user = getUserById(id);
            if (user != null) {
                users.add(user);
            }
        }

        return users;
    }

}

File: sample-chat-kotlin/app/src/main/java/com/quickblox/sample/chat/kotlin/utils/qb/QbUsersHolder.kt
package com.quickblox.sample.chat.kotlin.utils.qb

import android.util.SparseArray
import com.quickblox.users.model.QBUser
import java.util.*

/**
 * Basically in your app you should store userList in database
 * And load userList to memory on demand
 * We're using runtime SpaceArray holder just to simplify app logic
 */
object QbUsersHolder {
    private var qbUserSparseArray: SparseArray<QBUser> = SparseArray()

    fun putUsers(users: List<QBUser>) {
        for (user in users) {
            putUser(user)
        }
    }

    private fun putUser(user: QBUser) {
        qbUserSparseArray.put(user.id, user)
    }

    fun getUsersByIds(usersIds: List<Int>): List<QBUser> {
        val users = ArrayList<QBUser>()
        for (id in usersIds) {
            val user = getUserById(id)
            user?.let {
                users.add(user)
            }
        }
        return users
    }

    fun getUserById(id: Int): QBUser? {
        return qbUserSparseArray.get(id)
    }
}